index.js

```diff
@@ -69,13 +70,30 @@ function copyFiles(args, config, callback) {
         return next(err);
       }
       var outName = path.join(outDir, dealWith(pathName, opts));
+	  var done = function(){  		
+	    mkdirp(path.dirname(outName), function (err) {
+	      if (err) {
+	        return next(err);
+	      }
+	      next(null, pathName);
+	    });
+	  }
       if (pathStat.isFile()) {
-        mkdirp(path.dirname(outName), function (err) {
-          if (err) {
-            return next(err);
-          }
-          next(null, pathName);
-        });
+      	if(soft){
+		  fs.stat(outName, function(err, outStat){
+		  	if(err === null){
```

Comment:
you should use `!err`, it probably won't be undefined but no reason to to be so specific